zerver/webhooks/zapier/view.py

```diff
@@ -1,15 +1,32 @@
-from typing import Any, Dict
+from typing import Any, Dict, List
 
 from django.http import HttpRequest, HttpResponse
 from django.utils.translation import ugettext as _
 
 from zerver.decorator import api_key_only_webhook_view
 from zerver.lib.actions import check_send_private_message_from_emails
 from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.stream_subscription import get_stream_subscriptions_for_user
 from zerver.lib.response import json_error, json_success
 from zerver.lib.webhooks.common import check_send_webhook_message, \
     validate_extract_webhook_http_header
-from zerver.models import UserProfile
+from zerver.models import UserProfile, get_stream_by_id_in_realm, \
+    Stream
+
+def get_available_streams_for_bot_owner(bot: UserProfile) -> List[str]:
+    assert bot.bot_owner is not None
+    subs = get_stream_subscriptions_for_user(bot.bot_owner)
+    streams = []
+    for sub in subs:
+        stream_id = sub.recipient.type_id
+        try:
+            stream = get_stream_by_id_in_realm(stream_id, bot.realm)
```

Comment:
my guess is something like `Stream.objects.values_list('name').get(id=stream_id, realm=bot.realm)` will be faster than doing a `select_related` thing. (Though I'm also not sure why you need this check at all.)

Reply:
That won't work for the reason I outlined in my comment above. We don't have access to Stream objects, only Subscription objects. Also, I am not sure if the `Recipient` objects in the subscriptions also have recipients for other realms, so the `get_stream_by_id_in_realm` call could raise an exception for a stream that the user is subscribed to but which doesn't exist in the current realm. I could be getting all of this wrong though, I recently starting dabbling in this area of the code. @timabbott: Thoughts? :)

Reply:
Database queries in a loop are basically banned in the Zulip backend.  Don't do it.

You can do this with `do_get_streams` -- that's the thing that powers the `GET /streams` API endpoint.